src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -561,6 +561,36 @@ bool SQLStorage::loadDelegation(std::string* data, const Uptane::Role role) {
   return true;
 }
 
+bool SQLStorage::loadAllDelegations(DelegationArray& data) const {
+  bool result = false;
+
+  try {
+    SQLite3Guard db = dbConnection();
+
+    auto statement = db.prepareStatement("SELECT meta, role_name FROM delegations;");
+    auto statement_state = statement.step();
+
+    if (statement_state == SQLITE_DONE) {
+      LOG_TRACE << "Delegations metadata are not present";
+      return true;
+    } else if (statement_state != SQLITE_ROW) {
+      LOG_ERROR << "Can't get delegations metadata: " << db.errmsg();
+      return false;
+    }
+
+    do {
+      data.emplace_back(
+          DelegationRecord({statement.get_result_col_blob(0).value(), statement.get_result_col_str(1).value()}));
```

Comment:
This is missing the benefits of using `emplace_back` and doesn't compile on Xenial according to CI.

This is enough:

`data.emplace_back(statement.get_result_col_blob(0).value(), statement.get_result_col_str(1).value());`

Reply:
Yeah, your right, will fix it.